src/validateAtoms.ts

```diff
@@ -0,0 +1,89 @@
+import { atom, Getter, SetStateAction, WritableAtom } from 'jotai';
+import { loadable } from 'jotai/utils';
+
+import type { CommonState } from './atomWithValidate';
+
+type inferGeneric<Type> = Type extends AtomWithValidation<infer X> ? X : never;
+
+export type Validator<Keys extends symbol | string | number, Vals> = (
+  values: Record<Keys, Vals>,
+) => void | Promise<void>;
+
+export type ValidatorState = {
+  isValid: undefined | boolean;
+  error: null | Error | unknown;
+  isValidating: undefined | boolean;
+};
+
+type AtomWithValidation<Value> = WritableAtom<
+  CommonState<Value>,
+  SetStateAction<Value>
+>;
+
+type State<Values extends Record<string, unknown>> = {
```

Comment:
Should `unknown` be `any` like L28? I always wonder which is better, and how it's different.

Reply:
It's inferring the types already, 
anything should be fine on this type, i'd leave it to be `unknown` so people can add type checks 
L28 uses `any` because we need to infer the types from the passed atom group, adding `unknown` there would mean the dev would have to manually typecast each atom, which wouldn't be nice. 

as for using `unknown` here would force them to check types in the Validator which is needed if they are writing validations. Though, in most cases jotai is used with primitive types (number, strings,etc) the inference should already help with this.

Reply:
thanks for the explanation!